src/main/java/at/hannibal2/skyhanni/features/misc/TabWidgetSettings.kt

```diff
@@ -0,0 +1,63 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils.getInventoryName
+import at.hannibal2.skyhanni.utils.InventoryUtils.getUpperItems
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class TabWidgetSettings {
+
+    private val mainPageWidgetPattern = "^§7Currently:.*".toRegex()
+
+    private val subPageWidgetPattern = "^§eClick to .*".toRegex(RegexOption.IGNORE_CASE)
+
+    private val shownSettingPattern = "Shown .* Setting.*".toRegex(RegexOption.IGNORE_CASE)
+
+    @SubscribeEvent
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
```

Comment:
please use the InventoryFullOpenEvent to check the slot and save it into a map of slotId and color. 
Then call the BackgroundDrawEvent and just call this map, if null just don't highlight it.
And a boolean to enable the BackgroundDrawEvent ```if(!inInvenory) return```.
Last thing in the InventoryCloseEvent clear the ```inInvenory``` and the map.